### DISCUSSION TITLE
How to configure keycloak in apisix

### DISCUSSION THREAD

**Discussion by anjoy8** [apache/apisix#6088] [Q&A] [answered]
Hi,Hope to get help.
**I generated a token of keycloak, and the result after parsing is as follows:**

![image](https://user-images.githubusercontent.com/28941816/149288712-e739402a-a212-4b0d-afef-6e47a9757793.png)

**Then, in the route of apisix, configure the keycloak plug-in:**

![image](https://user-images.githubusercontent.com/28941816/149288883-61e423ea-4125-452a-ab83-e6e03edb4d81.png)

**Use this token to access this route. Unfortunately, there was an error:**
`{"error":"invalid_request","error_description":"Client does not support permissions"}`

![image](https://user-images.githubusercontent.com/28941816/149289001-02b192e2-cb65-4338-a368-011d9a293019.png)

**Comment by juzhiyuan** [comment]
The error shown in your screenshot is not from Apache APISIX IMO 🤔 Here is the authz-keycloack doc https://apisix.apache.org/docs/apisix/plugins/authz-keycloak/

And @starsz could you help to have a check?

**Reply by starsz** [reply]
> The error shown in your screenshot is not from Apache APISIX IMO 🤔 Here is the authz-keycloack doc https://apisix.apache.org/docs/apisix/plugins/authz-keycloak/
> 
> And @starsz could you help to have a check?

OK.Let me have  a check

**Comment by starsz** [comment, marked as answer]
Hi @anjoy8.Please make sure you have enabled the Authorization in keycloak client.

It's in Client-Settings

![image](https://user-images.githubusercontent.com/25628854/149702574-3892e693-1554-4713-90d8-e9d4c498c36d.png)

**Reply by anjoy8** [reply]
Yes, I have configured it
![image](https://user-images.githubusercontent.com/28941816/149704129-19d8fd01-b124-4380-a263-97476241b5cc.png)

**Reply by anjoy8** [reply]
Excuse me, do I need configuration here?
and how?

![image](https://user-images.githubusercontent.com/28941816/149704508-de6c8034-3841-48ac-84f0-0a3fc0bfa0d2.png)

**Reply by starsz** [reply]
> Excuse me, do I need configuration here? and how?
> 
> ![image](https://user-images.githubusercontent.com/28941816/149704508-de6c8034-3841-48ac-84f0-0a3fc0bfa0d2.png)

No, it's no need.

In your APISIX route, the audience is `account`. 
And the client ID you have enabled Authorization is `insurance_app` ?

IMO, I think you should enable it in the `account` client.

**Reply by anjoy8** [reply]
cool!  That's why, thank you very much.

By the way, what is the relationship between **account** and custom client(e.g. **insurance_app**)?
I find the **account** is generated by the system.

**Reply by starsz** [reply]
> cool! That's why, thank you very much.
> 
> By the way, what is the relationship between **account** and custom client(e.g. **insurance_app**)? I find the **account** is generated by the system.

Haha. Maybe you can refer to the keycloak doc: https://www.keycloak.org/docs/latest/server_admin/#assembly-managing-clients_server_administration_guide
